Update tag focus and select states to new design

mkernohanbc opened the pull request:
This PR implements the new offset-ring focus state design on the tag component.

This change makes it easier to distinguish when a tag is in focus, selected or both:

<img width="680" alt="Screenshot 2024-06-25 at 13 55 35" src="https://github.com/bcgov/design-system/assets/135075821/eb80c658-d13c-4acc-a8a6-b0cbb42fc1b8">

[New design in Figma](https://www.figma.com/design/2OhHsH2ztSFUwDcVzxug7Z/BC-Design-System?node-id=3347-129&m=dev), for reference.

ty2k commented:
The new focus style looks great!

One existing issue with this component is that our current markup and styles will cause some weirdness for the "Selection" story, as shown in your screenshot. The existing markup and style for the non-selected tag is causing its text and icon to sit visually higher in the container than it should be.

Here's a zoomed in example of what's in this branch currently:
<img width="1840" alt="Selected TagGroup using this branch's code" src="https://github.com/user-attachments/assets/a2c75901-ab78-4ef7-9b0e-1b27dc399f96">

I've added a [branch here](https://github.com/bcgov/design-system/tree/feature/refactor-tag-markup) where I've removed an unnecessary `<div>` from the markup and simplified the styling. This is what that version of the code looks like:
<img width="1840" alt="Selection TagGroup with updated markup" src="https://github.com/user-attachments/assets/8837df20-4424-40a0-849c-c841e45c5b1b">

If you open those two screenshots in separate tabs and switch between them, the visual difference should be obvious.

Does that change work for you instead? Can we cherry-pick that commit into this branch?

mkernohanbc commented:
> The new focus style looks great!
> 
> One existing issue with this component is that our current markup and styles will cause some weirdness for the "Selection" story, as shown in your screenshot. The existing markup and style for the non-selected tag is causing its text and icon to sit visually higher in the container than it should be.
> 
> Here's a zoomed in example of what's in this branch currently: <img alt="Selected TagGroup using this branch&apos;s code" width="1840" src="https://private-user-images.githubusercontent.com/25143706/349722154-a2c75901-ab78-4ef7-9b0e-1b27dc399f96.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNjAxNzYsIm5iZiI6MTcyMTI1OTg3NiwicGF0aCI6Ii8yNTE0MzcwNi8zNDk3MjIxNTQtYTJjNzU5MDEtYWI3OC00ZWY3LTliMGUtMWIyN2RjMzk5Zjk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDIzNDQzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5ZDc1MTc3MTE3NTVkNmE0ODIzZWM3MWE0Yjc3M2Q2NDMwYzJmZGRjMGMwNjFiOWU2NjU5YmVhOWUxMzJjNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5xfi39WZCsnx8ZgXAvRaNkFGcvAqvuee-3DaDh2ozsE">
> 
> I've added a [branch here](https://github.com/bcgov/design-system/tree/feature/refactor-tag-markup) where I've removed an unnecessary `<div>` from the markup and simplified the styling. This is what that version of the code looks like: <img alt="Selection TagGroup with updated markup" width="1840" src="https://private-user-images.githubusercontent.com/25143706/349722341-8837df20-4424-40a0-849c-c841e45c5b1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNjAxNzYsIm5iZiI6MTcyMTI1OTg3NiwicGF0aCI6Ii8yNTE0MzcwNi8zNDk3MjIzNDEtODgzN2RmMjAtNDQyNC00MGEwLTg0OWMtYzg0MWU0NWM1YjFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDIzNDQzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmMGMzMDBiM2RmMzY3ZDNmNzY0ZWFiN2I5YTc0OGQ0ZjBhYjY3OGY4OWEzZjMxZmU2NWQ5ODQ2OTVmYzMyMTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wduZVwtt5_7ZPwZ976soibXzbTEc8opJby7nCr7XeGU">
> 
> If you open those two screenshots in separate tabs and switch between them, the visual difference should be obvious.
> 
> Does that change work for you instead? Can we cherry-pick that commit into this branch?

Looks good to me! Merged in [18cf561](https://github.com/bcgov/design-system/pull/396/commits/18cf5619a182dcc305872d092c88163079728eff)
